src/rkopenmdao/butcher_tableau.py

```diff
@@ -11,17 +23,47 @@ def __init__(
         self,
         butcher_matrix: np.ndarray,
         butcher_weight_vector: np.ndarray,
-        butcher_time_stages: np.ndarray,
+        butcher_time_stages: np.ndarray | None = None,
+        p=None,
+        name='Runge-Kutta method',
     ):
-        self.butcher_matrix = butcher_matrix
+        
+        if len(np.shape(butcher_matrix)) == 2:
+            self.butcher_matrix = butcher_matrix
+        else:
+            self.butcher_matrix = np.array([butcher_matrix])
         self.butcher_weight_vector = butcher_weight_vector
-        self.butcher_time_stages = butcher_time_stages
+        if np.sum(self.butcher_weight_vector) != 1:
+            warnings.warn("Averaging weights do not sum up to 1")
+        if butcher_time_stages is None:
+            self.butcher_time_stages = np.sum(self.butcher_matrix, 1)
+        else:
+            self.butcher_time_stages = butcher_time_stages
+            if np.sum(self.butcher_matrix, 1) != butcher_time_stages:
```

Comment:
Comparison operations on numpy-arrays yield a numpy array made of booleans. You want to have np.all here.

Reply:
I think np.any is more appropriate since np.all is combinations of 'or' and we need combination of 'and' of the indices

Reply:
changed to:    
`if not np.all(np.sum(self.butcher_matrix, 1) == butcher_time_stages):`